integrations/wiz/client.py

```diff
@@ -0,0 +1,240 @@
+        }
+
+        if self.token_url not in audience_mapping:
+            raise Exception("Invalid Token URL")
+
+        audience = audience_mapping[self.token_url]
+
+        return {
+            "grant_type": "client_credentials",
+            "audience": audience,
+            "client_id": self.client_id,
+            "client_secret": self.client_secret,
+        }
+
+    def get_token(self) -> str:
+        try:
+            if cached_token_object := event.attributes.get(CacheKeys.TOKENS):
+                if not self.is_token_expired(cached_token_object):
+                    return cached_token_object["access_token"]
+
+            response = httpx.post(
+                self.token_url,
+                data=self.auth_params,
+                headers={"Content-Type": "application/x-www-form-urlencoded"},
+            )
+            response.raise_for_status()
+
+            token_result = response.json()
+
+            # Calculate and add expiry date to the token_result
+            expiry_in_seconds = token_result.get("expires_in", 0)
+            expiry_datetime = datetime.utcnow() + timedelta(seconds=expiry_in_seconds)
+            token_result["expiry_date"] = expiry_datetime.isoformat()
+
+            event.attributes[CacheKeys.TOKENS] = token_result
+
+            return token_result["access_token"]
+        except Exception as e:
+            print(e)
+            raise
+
+    def is_token_expired(self, token_object: Any) -> bool:
+        expiry_date_str = token_object.get("expiry_date")
+        if expiry_date_str:
+            expiry_date = datetime.fromisoformat(expiry_date_str)
+            return expiry_date <= datetime.utcnow()
+        return True
+
+    async def get_issues(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+        logger.info("Fetching issues from Wiz API")
+
+        if cache := event.attributes.get(CacheKeys.ISSUES):
+            logger.info("picking Wiz issues from cache")
+            yield cache
+
+        json_data: dict[str, Any] = {
+            "query": ISSUES_GQL,
+            "variables": {
+                "first": PAGE_SIZE,
+                "orderBy": {"direction": "DESC", "field": "CREATED_AT"},
+            },
+        }
+
+        try:
+            response = await self.http_client.post(
+                url=self.api_url,
+                json=json_data,
+                headers=self.api_auth_header,
+            )
+            response.raise_for_status()
+            response_json = response.json()
+            data = response_json.get("data")
+
+            if not data:
+                logger.error(response_json.get("error"))
+                raise Exception("No data found for Wiz account")
+
+            issues = data["issues"]["nodes"]
+            event.attributes[CacheKeys.ISSUES] = issues
+            yield issues
+
+        except httpx.HTTPError as e:
+            logger.error(f"Error while fetching issues: {str(e)}")
```

Comment:
why are u ignoring the exception? please raise it back

Reply:
I have raised it now